pkg/version/version.go

```diff
@@ -168,3 +171,13 @@ func latestSHA(tags []api.ImageTag) (*api.ImageTag, error) {
 
 	return latestTag, nil
 }
+
+func filterByOSAndArchitecture(tags []api.ImageTag, arch architecture.NodeMetadata) ([]api.ImageTag, error) {
+	var filteredTags = make([]api.ImageTag, len(tags))
+	for _, tag := range tags {
+		if tag.OS == arch.OS && tag.Architecture == arch.Architecture {
+			filteredTags = append(filteredTags, tag)
+		}
+	}
+	return filteredTags, nil
+}
```

Comment:
We are needlessly moving around a _lot_ of data here which is quite expensive.

Change this to `osArchMatch(tag api.ImageTag, os api.OS, arch api.Arch) bool`, and call it in both `latestSemvar` and `latestSHA`.

Reply:
We are not exposing the os and arch on every client yet, so we will need to update that first before merging this.

Without that fix, we get the following behaviour:

```
version_checker_is_latest_version{architecture="",container="cert-manager",current_version="v1.0.1",image="quay.io/jetstack/cert-manager-cainjector",latest_version="",namespace="cert-manager",os="",pod="cert-manager-cainjector-678b44fb8c-8w48j"} 1
version_checker_is_latest_version{architecture="",container="cert-manager",current_version="v1.0.1",image="quay.io/jetstack/cert-manager-controller",latest_version="",namespace="cert-manager",os="",pod="cert-manager-65b665b74d-lrbsv"} 1
version_checker_is_latest_version{architecture="",container="cert-manager",current_version="v1.0.1",image="quay.io/jetstack/cert-manager-webhook",latest_version="",namespace="cert-manager",os="",pod="cert-manager-webhook-57d5ffc9cb-srbd5"} 1
```

Reply:
We should also default to allow any os and arch if none are given in the options.

Meaning if os is nil in the options, no tags should be filtered on their os. Same with arch.